runtime/jcl/common/jdk_internal_foreign_abi_UpcallStubs.cpp

```diff
@@ -78,6 +76,7 @@ Java_jdk_internal_foreign_abi_UpcallStubs_freeUpcallStub0(JNIEnv *env, jobject r
 			J9UpcallMetaDataEntry * result = (J9UpcallMetaDataEntry *)hashTableFind(metaDataHashTable, &metaDataEntry);
 			if (NULL != result) {
 				J9UpcallMetaData *metaData = result->upcallMetaData;
+				J9Heap *thunkHeap = metaData->thunkHeapWrapper->heap;
```

Comment:
Regarding line 96 below: what prevents some other thread from failing with an execution exception while write-protection is disabled?

Reply:
How does the protection work? Is it per-thread or process-wide?